### PR TITLE
Adds foreign assets erc20 precompile article

### PR DESCRIPTION
**Pull request by augustocollado**
### Description

This PR adds the foreign assets precompile article

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] I have run my changes through Grammarly
- [x] If this page requires a disclaimer, I have added one
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/toolkit/ethereum-api/precompiles/foreign-assets-erc20.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+--8<-- 'text/builders/toolkit/ethereum-api/precompiles/security-note.md'
+
+## Prerequisites {: #prerequisites }
+
+ Tto follow along with the contents in this guide, you'll need:
+ 
+- Access to a Tanssi EVM appchain running [runtime 500](https://github.com/moondance-labs/tanssi/releases/tag/runtime-500){target=\_blank} or above
+- An established bidirectional XCM channel to another chain. To manage your appchain's channels, refer to the [Manage Cross-Chain Communication Channels](/builders/manage/dapp/xcm-channels/){target=\_blank} article
+- A registered foreign asset. Once the XCM channels are open, asset registration can be easily done using the [dApp](https://apps.tanssi.network/){target=\_blank}, in the `Asset Registration` menu entry from the `XCM` management section
```
[the remark this answers is not in the document]
Comment (augustocollado) [reply builders/toolkit/ethereum-api/precompiles/foreign-assets-erc20.md:33]: Yes, while writing this section I realized that we needed that guide. 
There's already a ticket to come back here and add the article once the guide is also merged -> https://opslayer.atlassian.net/jira/software/c/projects/DT/boards/108?selectedIssue=DT-253

**Comment by augustocollado** [comment]
> It looks good and comprehensive. I left some minor feedback.
> 
> I wonder if the interaction with Remix is way to detailed. I think it was great as you went through all the functions, but this makes the page a bit unnecessarily long. Wdyt?

Indeed, we could leave only:

Remix Set Up
Compile the Contract
Access the Contract
Get Basic Token Information
Get Account Balance
Send Transfer

And remove: 

Approve a Spend
Get Allowance of Spender
Send Transfer From Specific Account

**Comment by albertov19** [comment]
> > It looks good and comprehensive. I left some minor feedback.
> > I wonder if the interaction with Remix is way to detailed. I think it was great as you went through all the functions, but this makes the page a bit unnecessarily long. Wdyt?
> 
> Indeed, we could leave only:
> 
> Remix Set Up Compile the Contract Access the Contract Get Basic Token Information Get Account Balance Send Transfer
> 
> And remove:
> 
> Approve a Spend Get Allowance of Spender Send Transfer From Specific Account

Yeah I agree, let's trim this a bit

**Review by albertov19 — DISMISSED** [review]

**Review by albertov19 — DISMISSED** [review]
LGTM

File: builders/toolkit/ethereum-api/precompiles/external-assets-erc20.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    ![Import Tokens from Tokens Tab in MetaMask](/images/builders/toolkit/ethereum-api/precompiles/external-assets-erc20/external-assets-erc20-2.webp)
+
+Before continuing, you'll need the token's address, which, considering that in this example the external asset has an ID of `1`, will be:
+
+```text
+{{networks.dancebox.precompiles.external_assets_erc20_example}}
+```
+
+1. Enter the precompile address for the token contract address. When you enter the address, the **Token Symbol** and **Token Decimal** fields should automatically populate. If they do not, you can enter `UNIT` for the symbol and `12` for the decimal places.
```
Comment (eshaben) [inline builders/toolkit/ethereum-api/precompiles/external-assets-erc20.md:73]:
```suggestion
1. Enter the precompile address for the token contract address. When you enter the address, the **Token Symbol** and **Token Decimal** fields should automatically populate. If they do not, you can enter `UNIT` for the symbol and `12` for the decimal places
```

File: builders/toolkit/ethereum-api/precompiles/external-assets-erc20.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+### Access the Contract {: #access-the-contract }
+
+Instead of deploying the smart contract, you will access the interface through the address of external asset precompile:
+
+1. Click on the **Deploy and Run** tab directly below the **Compile** tab in Remix. Please note that the precompiled contracts are already accessible at their respective addresses. Therefore, there is no deployment step
+2. Make sure **Injected Web3** is selected in the **ENVIRONMENT** dropdown. Once you select **Injected Web3**, you may be prompted by MetaMask to connect your account to Remix if it's not already connected
+3. Make sure the correct account is displayed under **ACCOUNT**
+4. Ensure **IERC20 - IERC20.sol** is selected in the **CONTRACT** dropdown. Given that it is a precompiled contract, there is no deployment step. Instead, you are going to provide the address of the precompile in the **At Address** field
+5. Provide the address of the ERC-20 precompile (which is `{{networks.dancebox.precompiles.external_assets_erc20_example}}` in this example) and click **At Address**
+6. The **IERC20** precompile will appear in the list of **Deployed Contracts**.
```
Comment (eshaben) [inline builders/toolkit/ethereum-api/precompiles/external-assets-erc20.md:113]:
```suggestion
6. The **IERC20** precompile will appear in the list of **Deployed Contracts**
```

**Review by eshaben — APPROVED** [review]
LGTM!

**Pull request merged**